src/packets/caching_sha2_password.rs

```diff
@@ -0,0 +1,75 @@
+use std::io;
+
+use crate::{
+    io::ParseBuf,
+    proto::{MyDeserialize, MySerialize},
+};
+
+define_header!(
+    PublicKeyRequestHeader,
+    InvalidPublicKeyRequest("Invalid PublicKeyRequest header"),
+    0x02
+);
+
+/// A request for a public RSA key, used by some authentication mechanisms
+/// to add a layer of protection to an unsecured channel.
```

Comment:
This is just to clarify the meaning (there is a lot of possible steps and packets in caching_sha2_password)

```suggestion
/// A client request for a server public RSA key, used by some authentication mechanisms
/// to add a layer of protection to an unsecured channel (see [`PublicKeyResponse`]).
///
/// [`PublicKeyResponse`]: crate::packets::PublicKeyResponse
```